examples/diff.c

```diff
@@ -335,3 +357,105 @@ static void diff_print_stats(git_diff *diff, struct opts *o)
 	git_buf_free(&b);
 	git_diff_stats_free(stats);
 }
+
+void die(char* message){
+	fprintf(stderr, message);
+	exit(EXIT_FAILURE);
+}
+
+struct pair extract_from_string(const char *cmd_opt)
+{
+	struct pair p;
+	char **endptr;
+	const char * cmd_opt2;
+	int tmp = strtol(cmd_opt,endptr,'A');
+
+
+	if (tmp==0) {
```

Comment:
`sscanf` would be another option. But even if it's not that, this code needs to be refactored into some helper function or loop as it has multiple copies of the same parsing code in different branches.